pyproject.toml

```diff
@@ -111,7 +111,7 @@ full-no-qt = [
   "nilearn",
   "numba",
   "openmeeg >= 2.5.5",
-  "pandas",
+  "pandas",  # >= 1.5
```

Comment:
Also I noticed in our unification of README and pyproject etc. in #12890 we didn't document anywhere what our internal min (for whichever release is coming up next) is AFAIK. We used to have:
```
- `pandas <https://pandas.pydata.org>`__ ≥ 1.5.2
```
Here I've put `pandas` in there. We could consider explicit pins at some point, but for now comments seem okay. For now I've only added `pandas` since it's the only one where I removed a shim for < 1.5.

Reply:
what's the downside to putting an explicit (non-comment) pin right now?  Is it that the repo README will reflect pins that don't apply to current stable release? (doesn't seem that bad to me... when we add new deps we add them into `pyproject.toml` right away in the PR where they're needed so in general I expect the repo README to reflect `main` not PyPI 🤷🏻)

Reply:
Just that in most cases we are 99% compatible with older versions, and just a single test or two breaks, or some niche plotting function. But it's probably most standard to add an explicit lower bound so I'll do that